File: yh_turtle/scripts/turtle_keyboard_clear.py
#!/usr/bin/python
#-*- coding: utf-8 -*-

import rospy
from geometry_msgs.msg import Twist
from turtle_clear import TurtleClear

class MyTurtle(TurtleClear):
    def __init__(self):
        super().__init__()
        self.pub = rospy.Publisher("turtle1/cmd_vel",Twist,queue_size=20)
        self.sub = rospy.Subscriber("turtle1/cmd_vel",Twist,self.msgCallback)

    def msgCallback(self,msg):
        if msg.linear.z > 0:
            self.call()
        else:
            self.pub.publish(msg)

rospy.init_node("turtle_keyboard_clear")
my_turtle = MyTurtle()
rospy.spin()

File: yh_turtle/src/turtle_keyboard_clear.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"
#include "yh_turtle/my_turtle.h"

int main(int argc, char** argv)
{
    ros::init(argc, argv, "turtle_keyboard_clear");
    MyTurtle my_turtle;

    ros::spin();

    return 0;
}
